src/BatteryStateOfChargeService.cpp

```diff
@@ -43,21 +47,43 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return totalAmpUsed_;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
-    return false;
+    if (newCurrent_ < 0)
+    {
+        return true;
+    }
+    else
+    {
+        return false;
+    }
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+    double seconds = QTime(0,0,0).secsTo(newTime_);
+    double timeelapsed =(BATTERY_AMP_HOUR_CAPACITY / fabs(newCurrent_)) * SECONDS_TO_HOURS;
```

Comment:
Sorry I guess I should've been clearer about naming. Local variables should follow camelCase convention. Where the first letter of each word is capitalized, except for the first word. Some other types of naming conventions are cap case and snake case. Here are some examples.

`CAP_CASE` mainly for constants.
`IRS_TIME_DONE`
`CONST`

`snake_case` is a standard for python and is becoming more and more prevalent in C++
`an_example`
`irs_time`